### ISSUE TITLE
 Closed Uncaught (in promise) when pdfjs canva in bootstrap caroussel

### ISSUE THREAD

**Issue by jbonlinea** [mozilla/pdf.js#13677]
This is basically a narrowed down issue first mentionned #13663 

I provide an example, which rely on pdfjs learning example hello world.

what happens is that when pdfjs canva is visible on load, the pdf is proprely displayed, but when the canva is not displayed the errors pasted bellow appears.

This example use bootstrap accordion, one accordion item is closed on load, and hte pdf within fails.
You may open and close accordion quickly while the page is still loading, then there will be no errors and all pdf will be nicely loaded

Configuration:

Static hostname: debian
Icon name: computer-laptop
Chassis: laptop
Operating System: Debian GNU/Linux 9 (stretch)
Kernel: Linux 4.9.0-16-amd64
Architecture: x86-64

Steps to reproduce the problem:

use the code provided with your own working hello world

```
<!doctype html>
<html lang="en">
  <head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <!-- Bootstrap CSS -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">

    <title>Hello, world!</title>
  </head>
  <body>


<div class="accordion" id="accordionExample">
  <div class="accordion-item">
    <h2 class="accordion-header" id="headingOne">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
        Accordion Item #1
      </button>
    </h2>
    <div id="collapseOne" class="accordion-collapse collapse show" aria-labelledby="headingOne" data-bs-parent="#accordionExample">
      <div class="accordion-body">
        <strong>This is the first item's accordion body.</strong> It is shown by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions. You can modify any of this with custom CSS or overriding our default variables. It's also worth noting that just about any HTML can go within the <code>.accordion-body</code>, though the transition does limit overflow.
         <iframe src="helloworld.html" style="width:350px; height: 350px"></iframe>      
      </div>
    </div>
  </div>
  <div class="accordion-item">
    <h2 class="accordion-header" id="headingTwo">
      <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseTwo" aria-expanded="false" aria-controls="collapseTwo">
        Accordion Item #2
      </button>
    </h2>
    <div id="collapseTwo" class="accordion-collapse collapse" aria-labelledby="headingTwo" data-bs-parent="#accordionExample">
      <div class="accordion-body">
        <strong>This is the second item's accordion body.</strong> It is hidden by default, until the collapse plugin adds the appropriate classes that we use to style each element. These classes control the overall appearance, as well as the showing and hiding via CSS transitions. You can modify any of this with custom CSS or overriding our default variables. It's also worth noting that just about any HTML can go within the <code>.accordion-body</code>, though the transition does limit overflow.

         <iframe src="helloworld.html" style="width:350px; height: 350px"></iframe>      
      </div>
    </div>
  </div>
 
</div>

         <iframe src="helloworld.html" style="width:350px; height: 350px"></iframe>      


    <!-- Optional JavaScript; choose one of the two! -->

    <!-- Option 1: Bootstrap Bundle with Popper -->
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>

  </body>
</html>

```

What is the expected behavior? (add screenshot)
all pdfjs canva to load the pdf file properly

What went wrong? (add screenshot)
```


Uncaught (in promise) 
Exception
​
columnNumber: 0
​
data: null
​
filename: "http://dev.mysite.fr/pdf.js-master/node_modules/pdfjs-dist/build/pdf.js"
​
lineNumber: 13363
​
message: ""
​
name: "NS_ERROR_FAILURE"
​
result: 2147500037
​
stack: "resetCtxToDefault@http://dev.mysite.fr/pdf.js-master/node_modules/pdfjs-dist/build/pdf.js:13363:5\nbeginDrawing@http://dev.mysite.fr/pdf.js-master/node_modules/pdfjs-dist/build/pdf.js:13492:24\ninitializeGraphics@http://dev.mysite.fr/pdf.js-master/node_modules/pdfjs-dist/build/pdf.js:11729:18\nrender/<@http://dev.mysite.fr/pdf.js-master/node_modules/pdfjs-dist/build/pdf.js:9918:28\n"
​
<prototype>: ExceptionPrototype
​​
columnNumber: 
​​
data: 
​​
filename: 
​​
lineNumber: 
​​
message: 
​​
name: 
​​
result: 
​​
stack: 
​​
toString: function toString()
​​
Symbol(Symbol.toStringTag): "Exception"
​​
<get columnNumber()>: function columnNumber()
​​
<get data()>: function data()
​​
<get filename()>: function filename()
​​
<get lineNumber()>: function lineNumber()
​​
<get message()>: function message()
​​
<get name()>: function name()
​​
<get result()>: function result()
​​
<get stack()>: function stack()
​​
<set stack()>: function stack()
​​
<prototype>: Object { … }
canvas.js:787:4

```

I guess it's an edge case, but any div that is not display on load but only after page load might encounter the same issue
Hope this may help

Regards

**Comment by Snuffleupagus** [comment]
Please see https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us in the Matrix room -- **keep in mind that just code snippets won't help us troubleshoot the problem**.

**Comment by jbonlinea** [comment]
Hi 

sorry about that, I though that using your own working pdfjs canvas could help to narrow the issue in this case as therre are iframes, buthere is the example provided with online demo, the only edit is to set the proper path name and create this dual.html file


http://test.quagga.fr/pdf.js/examples/learning/
http://test.quagga.fr/pdf.js/examples/learning/dual.html

regards

**Comment by jbonlinea** [comment]
Further example with of the exact same built with uikit instead of bootstrap 

here the pdf are displayed correctly :)
http://test.quagga.fr/pdf.js/examples/learning/dual-uikit.html

**Comment by Snuffleupagus** [comment]
Unfortunately none of the links above work any more, hence closing as INCOMPLETE for now.
